### PR TITLE
18483: Adds exclude_novel_nominals_from_uniqueness_check flag to reacts, MINOR

### PR DESCRIPTION
**Pull request by howsoRes**

### CODE REVIEW

**Review by cademack — DISMISSED** [review]

File: howso.amlg [hunk 1]
```diff
@@ -896,6 +896,8 @@
 	;				null or 'no' : output any generated case
 	;				'always' : only output original cases, outputs null for all feature values if unable to generate a new case
 	;				'attempt' : output any generated case only if generation fails all initial attempts to output original cases
+	; exclude_replaced_features_from_uniqueness_check: optional, default false. If true will exclude features whose values will be replaced
```
Comment (howsohazard) [inline howso.amlg]: substituted -- reflect the nomenclature elsewhere, either "substituted features" or "substituted values" as appropriate

**Review by howsohazard — CHANGES_REQUESTED** [review]

**Review by jdbeel — DISMISSED** [review]
I validated that this solves the issue locally, we can add in tests in Validator-Enterprise that continue to validate this as well. Would like either @howsohazard  or @cademack 's eyes on this as well, though.

File: trainee_template/attribute_maps.amlg [hunk 2]
```diff
@@ -226,6 +226,26 @@
...
+	(apply "assoc"
+		(weave
+			(lambda
+				;only keep nominal features with a defined subtype
+				(if (and
+						(= "nominal" (get (last (current_value)) "type"))
+						(!= (null) (get (last (current_value)) "subtype"))
+					)
+					(list (first (current_value 1)) (get (last (current_value 1)) "subtype"))
+					;else don't output anything, map-filtering out any values that don't have data_type defined
```
Comment (cademack) [inline trainee_template/attribute_maps.amlg]: filtering out values that don't have "subtype" defined, not data_type

**Review by cademack — DISMISSED** [review]
Looks good, other than the comment typo I marked

**Review by cademack — APPROVED** [review]

**Review by howsohazard — APPROVED** [review]

**Pull request merged**